components/pluralrules/src/operands.rs

```diff
@@ -200,3 +222,59 @@ macro_rules! impl_signed_integer_type {
 
 impl_integer_type!(u8 u16 u32 u64 u128 usize);
 impl_signed_integer_type!(i8 i16 i32 i64 i128 isize);
+
+impl From<&FixedDecimal> for PluralOperands {
+    /// Converts a [icu_num_util::FixedDecimal] to [PluralOperands]
+    fn from(f: &FixedDecimal) -> Self {
+        let mag_range = f.magnitude_range();
+        let mag_end = *mag_range.end();
+        let integer_range = 0..=mag_end;
+        let mag_start = *mag_range.start();
+
+        let mut integer_part: u64 = 0;
+        for magnitude in integer_range.rev() {
+            integer_part *= 10;
+            integer_part += f.digit_at(magnitude) as u64;
+        }
+
+        // The fractional part of the number, expressed as an integer to preserve trailing zeroes.
+        // For example, could be "100" if the stored number is "0.100".
+        let mut fraction_part_full: u64 = 0;
+        // A running total of the number of trailing zeros seen in the fractional part of the
+        // number.
+        let mut num_trailing_zeros = 0;
```

Comment:
If it would make it easier, you could add a function to FixedDecimal named `magnitude_range_nonzero(&self)` that behaves like `magnitude_range(&self)` except that it skips leading and trailing zeros.  This should be easy to calculate within FixedDecimal based on `magnitude` and the length of the digits array.

Reply:
I don't think it matters, benchmark-wise.